src/main/java/com/dislinkt/post/controller/PersonController.java

```diff
@@ -0,0 +1,47 @@
+package com.dislinkt.post.controller;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.dislinkt.post.dto.ErrorDTO;
+import com.dislinkt.post.dto.PersonDTO;
+import com.dislinkt.post.dto.ResponseDTO;
+import com.dislinkt.post.service.PersonService;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/people")
+public class PersonController {
+    
+    @Autowired
+    private PersonService service;
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<ResponseDTO<List<PersonDTO>>> getAll(){
+        ResponseDTO<List<PersonDTO>> ret = new ResponseDTO<>(service.findAll());
+        return new ResponseEntity<>(ret, HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity addPerson(@RequestHeader("id") UUID id, @RequestBody PersonDTO dto){
```

Comment:
You need to add @Valid here before @RequestBody to make validators work. Sorry about that, forgot to mention.

Reply:
No problem. Just tried it and I think I will just use these manual checks in the future instead of validators because I can send a responese with custom class ErrorDTO instead of a server error as this will give the user a clearer message

Reply:
Hmm I understand what you mean, just looked for solution around that. You can use this @ExceptionHandler, see example on this site https://www.baeldung.com/spring-boot-bean-validation.

Reply:
Thank you, I enter and changed the given example and now errors can be caught without the specific if clauses in methods.